hs_core/tasks.py

```diff
@@ -129,4 +131,78 @@ def add_zip_file_contents_to_resource(pk, zip_file_path):
         logger.error(exc_info)
     finally:
         # Delete upload file
-        os.unlink(zip_file_path)
\ No newline at end of file
+        os.unlink(zip_file_path)
+
+
+@shared_task
+def create_bag_by_irods(resource_id, istorage=None):

```

Comment:
@pkdash Unit test for this function already exists here: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/tests/api/native/test_bagit.py#L48..L53. I don't see MockIRODSTestCaseMixin is imported and used here in test_bagit.py, so I assume it is not mocked. This function already exists as an utility function and what I did is to move it to an async celery task, so previous unit test for it should still apply.